### PR TITLE
fix(vite): prefer HMR `ws` protocol

### LINKED ISSUE #4191: Infinite live reload when --https is active
### Environment

Nuxt CLI v3.0.0-27489614.315e5b5
RootDir: /Users/supianidz/Sites/foo
Nuxt project info:                                                                                          
------------------------------
- Operating System: `Darwin`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0-27489614.315e5b5`
- Package Manager:  `yarn@1.23.0-20220130.1630`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

1. npx nuxi init foo
2. cd foo
3. yarn install
4. generate certificate using mkcert
5. yarn dev --https --ssl-cert ./dev.crt --ssl-key ./dev.key

### Describe the bug

When the `--https` parameter is included along with the certificate from mkcert in the `yarn dev` command, nuxt3 will reload the page in a few seconds without stopping. So I have to run `yarn build` when I want to test the page without being bothered by weird reloads.

This is very annoying when dealing with form fields.

But it will all be normal if I don't include `--https`

https://user-images.githubusercontent.com/37969970/162358912-ecba2c1b-a5af-484b-84e9-cdfad08c584e.mp4

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by Diizzayy**
Closes #4191

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR is meant to allow seamless HMR connection in instances where vite's `https` mode may be enforced, by defaulting the HMR protocol to `ws`.

Though in the use case where a user's development server is served via https behind a reverse proxy (ie. nginx, etc...), note that the proxy config as well as vite's HMR config would need to be tweaked to allow proper usage in said setup.

Detailed explanation for using [nuxt dev server behind a NGINX Proxy with SSL](https://github.com/nuxt/framework/issues/1021#issuecomment-947123482)

@pi0 I think I should add this nginx use case to the documentation, as I myself and surely others utilize it.

### CODE REVIEW

File: packages/vite/src/vite.ts [hunk 1]
```diff
@@ -106,7 +107,8 @@ export async function bundle (nuxt: Nuxt) {
       .then(() => logger.info(`Vite ${env.isClient ? 'client' : 'server'} warmed up in ${Date.now() - start}ms`))
       .catch(logger.error)
   })
-
   await buildClient(ctx)
   await buildServer(ctx)
+
+  console.log('vite ctx', ctx.config.server.hmr)
```
Comment (Rigo-m) [inline packages/vite/src/vite.ts:113]: Careful about console log

Comment (Diizzayy) [reply packages/vite/src/vite.ts:113]: oops!

Comment (Diizzayy) [reply packages/vite/src/vite.ts]: @Rigo-m Thanks for pointing that out, I could've sworn I removed it lol.

**Comment by misaon** [comment]
Good point @Diizzayy

File: packages/bridge/src/vite/vite.ts [hunk 2]
```diff
@@ -106,6 +106,7 @@ async function bundle (nuxt: Nuxt, builder: any) {
             ignored: isIgnored
           },
           hmr: {
+            protocol: 'ws',
```
Comment (pi0) [inline packages/bridge/src/vite/vite.ts:109]:
```suggestion
            // https://github.com/nuxt/framework/issues/4191
            protocol: 'ws',
```

File: packages/vite/src/vite.ts [hunk 3]
```diff
@@ -75,6 +75,7 @@ export async function bundle (nuxt: Nuxt) {
             ignored: isIgnored
           },
           hmr: {
+            protocol: 'ws',
```
Comment (pi0) [inline packages/vite/src/vite.ts:78]:
```suggestion
            // https://github.com/nuxt/framework/issues/4191
            protocol: 'ws',
```

**Comment by pi0** [comment]
Thanks!

**Pull request merged**